### PR TITLE
[Gst-va] add Vpp crop support

### PR DESCRIPTION
**Pull request by wenqingx**

### CODE REVIEW

**Review by uartie — CHANGES_REQUESTED** [review]
This is incomplete.  The crop parameters also need to be mapped to the gst-va command-line in `lib/gstreamer/va/vpp.py:gen_vpp_opts(...)`.  See `lib/gstreamer/vaapi/vpp.py` for an example.

File: test/gst-va/vpp/crop.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+@slash.requires(*platform.have_caps("vpp", "crop"))
+class default(VppTest):
+  def before(self):
+    vars(self).update(
+      caps    = platform.get_caps("vpp", "crop"),
+      metric  = dict(type = "md5"),
+      vpp_op  = "crop",
+    )
+    super(default, self).before()
```
Comment (uartie) [inline test/gst-va/vpp/crop.py:21]: `super().before()` is better

**Comment by uartie** [comment]
> This is incomplete. The crop parameters also need to be mapped to the gst-va command-line in `lib/gstreamer/va/vpp.py:gen_vpp_opts(...)`. See `lib/gstreamer/vaapi/vpp.py` for an example.

Ah, actually the vapostproc plugin uses crop meta for crop support.  Thus, we need to insert a videocrop before the vapostproc.  This can be done inside the `lib/gstreamer/va/vpp.py:gen_output_opts(...)` function override.

**Comment by uartie** [comment]
https://gstreamer.freedesktop.org/documentation/va/vapostproc.html?gi-language=c#cropping-example

**Comment by uartie** [comment]
> > This is incomplete. The crop parameters also need to be mapped to the gst-va command-line in `lib/gstreamer/va/vpp.py:gen_vpp_opts(...)`. See `lib/gstreamer/vaapi/vpp.py` for an example.
> 
> Ah, actually the vapostproc plugin uses crop meta for crop support. Thus, we need to insert a videocrop before the vapostproc. This can be done inside the `lib/gstreamer/va/vpp.py:gen_output_opts(...)` function override.

Also, we should set `disable-passthrough=true` to force vapostproc to handle the cropping instead of passing to downstream.

**Comment by wenqingx** [comment]
thanks for review, and Focus help to do this part.closed. https://github.com/intel/vaapi-fits/pull/522

**Pull request closed without merging**